vr-xcon/xcon.act

```diff
@@ -1,8 +1,34 @@
 import net
 import time
+import file
 
-actor TcpEndpoint(connect_auth, dns, name, interface):
+actor Healthcheck(env, expected):
+    # keep track of TcpEndpoint actor states (ids, like "r1/1")
+    var states = {}
+
+    fa = file.FileAuth(env.auth)
+    wfa = file.WriteFileAuth(fa)
+
+    def update(endpoint, state):
+        states[endpoint] = state
+        _flush()
+
+    def _flush():
+        count = 0
+        for state in states.values():
+            if state == 3:
+                count += 1
+        print("healtcheck:", count == expected)
+        wf = file.WriteFile(wfa, "health")
+        health = "%d\nExpected %d sockets but only %d connected" % (0 if count == expected else 1, expected, count)
+        await async wf.write(health.encode())
+        await async wf.close()
```

Comment:
Just FYI, I guess you copied these lines verbatim from the test_file.act program. You don't need await async here, I think.

If there is no assignment of the return value, methods on other actors are called async and I think that's fine here. Nothing awaits on _flush, so there is little point for _flush to await the filewriter. I mean, one could imagine more error handling but we can't even catch the exception that WriteFile might raise so it's impossible right now...

Like see https://www.acton-lang.org/learn/acton-by-example/actors/async_method_call.html (feedback on  docs also welcome :) )

Reply:
Exactly, removed awaits.